sdk/contentsafety/azure-ai-contentsafety/src/main/java/com/azure/ai/contentsafety/ContentSafetyClientBuilder.java

```diff
@@ -283,8 +284,17 @@ private ContentSafetyClientImpl buildInnerClient() {
         return client;
     }
 
+    @Generated
+    private void validateBuilderParameters() {
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
```

Comment:
The change on `endpoint` would apply to any required client properties.

discussion
1. Should we use a different exception, e.g. `InvalidStateException`?
2. Do we put these "default validation" here, or in `createHttpPipeline`?

Reply:
`InvalidStateException` seems most correct to me.

I think I'm in favor of leaving all of the validation in this method. Perhaps we generate a comment that these are the default requirements and others can be added? "Magic" other validation seems hard to understand.